src/app/icd/client/DefaultCheckInDelegate.cpp

```diff
@@ -44,5 +44,44 @@ void DefaultCheckInDelegate::OnCheckInComplete(const ICDClientInfo & clientInfo)
 #endif
 }
 
+ICDRefreshKeyInfo * DefaultCheckInDelegate::OnKeyRefreshNeeded(const ICDClientInfo & clientInfo,
+                                                               const ICDClientStorage * clientStorage)
+{
+    auto icdRefreshKeyInfo = CreateICDRefreshKeyInfo(clientInfo, clientStorage);
+
+    if (icdRefreshKeyInfo == nullptr)
+    {
+        return nullptr;
+    }
+
+    CHIP_ERROR err = chip::Crypto::DRBG_get_bytes(icdRefreshKeyInfo->mNewKey.Bytes(), icdRefreshKeyInfo->mNewKey.Capacity());
+    if (err != CHIP_NO_ERROR)
+    {
+        return nullptr;
```

Comment:
This right here is a leak, if CreateICDRefreshKeyInfo actually returned a value we are supposed to free....  I don't understand the shape of these APIs; if we have unique ptrs, why are we writing leaky code by ignoring them?

Reply:
I was intially using Platform::New and then was trying out other things.